src/wwwroot/app/directives/candidate-vacancy-info/candidate-vacancy-info.js

```diff
@@ -29,179 +35,229 @@ export default class CandidateVacancyInfoDirective {
+   function _init() {
+      if (vm.parentEntity === 'vacancy') {
+         vm.rejectStages = filter(vm.vacancyStages, (extStage) => {
+            return extStage.stage.stageType === STAGE_TYPES.RejectStage;
          });
-      } else {
-         deffered.resolve({ comment: null,
-            isPassed: true,
-               stagesObject,
-               stageToPass  });
+         vm.stagesToShow = calculateVacancyStagesEntitiesCount(
+               vm.vacancyStageInfosComposedByCandidateIdVacancyId);
+      } else if (vm.parentEntity === 'candidate') {
+         vm.vacancyStageInfosComposedByCandidateIdVacancyId = map(
+               vm.vacancyStageInfosComposedByCandidateIdVacancyId, findAndSetRejectStagesFor);
+         if (some(vm.vacancyStageInfosComposedByCandidateIdVacancyId)) {
+            //TODO intersect of stages
+            vm.stagesToShow = calculateVacancyStagesEntitiesCount(vm.vacancyStageInfosComposedByCandidateIdVacancyId);
+         }
       }
-      return deffered.promise;
    }
 
-   function showSingleCommentDialog() {
+   function findAndSetRejectStagesFor(vacancyStage)  {
+      return Object.assign (vacancyStage, {
+         rejectStages: filter(vacancyStage.stageFlow, (extStage) => {
+            return extStage.stage.stageType === STAGE_TYPES.RejectStage;
+         })
+      });
+   }
+
+   function calculateVacancyStagesEntitiesCount(vacancyStageInfosComposedByCandidateIdVacancyId) {
+      let stagesFlow = vm.vacancyStage || head(vacancyStageInfosComposedByCandidateIdVacancyId).stageFlow;
+      let withoutHireStage = filter(stagesFlow, (extStage) => {
+         return extStage.stage.stageType !== STAGE_TYPES.HireStage;
+      });
+      return map(withoutHireStage, (extStage) => {
+         let stageEntitiesCount = reduce(vacancyStageInfosComposedByCandidateIdVacancyId, (count, vsi) => {
+            return vsi.currentStageId === extStage.stage.id ? count + 1 : count + 0;
+         }, 0);
+         return Object.assign(extStage, {
+            entitiesCount: stageEntitiesCount,
+            _hasEntities: stageEntitiesCount !== 0,
+            _isPressed: false
+         });
+      });
+   }
+   _init();
+
+   vm.callStagesDialog = (entityStageObject) => {
+      let stageFlow = entityStageObject.stageFlow ? entityStageObject.stageFlow : vm.vacancyStages;
+      showStagesFlowDialogFor(entityStageObject, stageFlow)
+         .then(changedVSIs => {
+            updateCandidateStagesForWith(entityStageObject, changedVSIs);
+         })
+         .then(notifySuccess)
+         .catch((notChangedVSIs) => {
+            updateCandidateStagesForWith(entityStageObject, notChangedVSIs);
+         });
+   };
+
+   function showStagesFlowDialogFor(entityStageObject, vacancyStages) {
       let dialogResult = $q.defer();
-      let dialogTransferObject = { comment: ' ' };
+      let mainStages = filter(vacancyStages, (extStage) => {
+         return extStage.stage.stageType === STAGE_TYPES.MainStage;
+      });
```

Comment:
it can be next:
let mainStages = filter(vacancyStages, extStage => extStage.stage.stageType === STAGE_TYPES.MainStage);